src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,391 @@
+
+    private static class TransferMeta {
+        public final TransferType type;
+        public final int upcharge;
+        public final boolean payOnExit;
+
+        /**
+         * Create a TransferMeta
+         * @param type Type of transfer
+         * @param upcharge Upcharge for the transfer in cents
+         * @param payOnExit Whether the fare is charged at end of leg
+         */
+        public TransferMeta(TransferType type, int upcharge, boolean payOnExit) {
+            this.type = type;
+            this.upcharge = upcharge;
+            this.payOnExit = payOnExit;
+        }
+
+        public TransferMeta(TransferType type) {
+            this(type, 0, false);
+        }
+    }
+
+    private static RideType classify(Ride ride) {
+        Route routeData = ride.routeData;
+        String shortName = routeData.getShortName();
+
+        switch(routeData.getId().getAgencyId()) {
+            case COBB_AGENCY_ID:
+                if (shortName.equalsIgnoreCase("100") ||
+                    shortName.equalsIgnoreCase("101") ||
+                    shortName.equalsIgnoreCase("102")) {
+                    return RideType.COBB_EXPRESS;
+                } else if (shortName.equalsIgnoreCase("BLUE") ||
+                    shortName.equalsIgnoreCase("GREEN")) {
+                    return RideType.FREE_RIDE;
+                }
+                return RideType.COBB_LOCAL;
+            case XPRESS_AGENCY_ID:
+                // Get hour of trip start
+                long hours = (ride.startTime / 60 / 60) % 24;
+                if (hours >= 12) {
+                    return RideType.XPRESS_AFTERNOON;
+                } else {
+                    return RideType.XPRESS_MORNING;
+                }
+            case GCT_AGENCY_ID:
+                if (shortName.equalsIgnoreCase("102") ||
+                    shortName.equalsIgnoreCase("103A") ||
+                    shortName.equalsIgnoreCase("110") ||
+                    shortName.equalsIgnoreCase("swpr")) {
+                    return RideType.GCT_EXPRESS_Z1;
+                } else if (shortName.equalsIgnoreCase("101") ||
+                    shortName.equalsIgnoreCase("103")) {
+                    return RideType.GCT_EXPRESS_Z2;
+                }
+                return RideType.GCT_LOCAL;
+            // Also catches MARTA_AGENCY_ID
+            default:
+                // Streetcar GTFS published by MARTA
+                if(shortName.equalsIgnoreCase("atlsc")) {
+                    return RideType.STREETCAR;
+                }
+                return RideType.MARTA;
+        }
+    }
+
+    private static int getMaxTransfers(RideType rideType) {
+        switch (rideType) {
+            // GCT only allows 3 transfers.
+            case GCT_EXPRESS_Z1:
+            case GCT_LOCAL:
+            case GCT_EXPRESS_Z2:
+                return 3;
+            default:
+                return 4;
+        }
+    }
+
+    private static int getTransferWindow(RideType rideType) {
```

Comment:
It seems unnecessary to have method with arguments here, doesn't it? A constant would do.

Reply:
Yes, but it's possible that in the future there might be different transfer windows depending on the rideType.